src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/GoogleDataprocDAO.scala

```diff
@@ -70,19 +81,133 @@ class GoogleDataprocDAO(protected val dataprocConfig: DataprocConfig)(implicit v
     }
   }
 
-
-  def deleteCluster(googleProject: String, clusterName: String)(implicit executionContext: ExecutionContext): Future[Unit] = {
+  override def deleteCluster(googleProject: String, clusterName: String)(implicit executionContext: ExecutionContext): Future[Unit] = {
     Future {
       //currently, the bucketPath of the clusterRequest are not used - it will be used later as a place to store notebooks and results
-      val request = dataproc.projects().regions().clusters().delete(googleProject, dataprocConfig.dataprocDefaultZone, clusterName)
+      val request = dataproc.projects().regions().clusters().delete(googleProject, dataprocConfig.dataprocDefaultRegion, clusterName)
       try {
         executeGoogleRequest(request)
       } catch {
-        case e:GoogleJsonResponseException =>
-          if(e.getStatusCode!=404)
+        case e: GoogleJsonResponseException =>
+          if (e.getStatusCode != 404)
             throw CallToGoogleApiFailedException(googleProject, clusterName, e.getStatusCode, e.getDetails.getMessage)
       }
     }
   }
 
+  override def getClusterStatus(googleProject: GoogleProject, clusterName: String)(implicit executionContext: ExecutionContext): Future[ClusterStatus] = {
+    getCluster(googleProject, clusterName).map { cluster =>
+      ClusterStatus.withNameIgnoreCase(cluster.getStatus.getState)
+    }
+  }
+
+  override def getClusterMasterInstanceIp(googleProject: GoogleProject, clusterName: String)(implicit executionContext: ExecutionContext): Future[Option[String]] = {
+    val ipOpt = for {
+      cluster <- OptionT.liftF[Future, Cluster] { getCluster(googleProject, clusterName) }
+      masterInstanceName <- OptionT.fromOption { getMasterInstanceName(cluster) }
+      masterInstanceZone <- OptionT.fromOption { getZone(cluster) }
+      masterInstance <- OptionT.liftF[Future, Instance] { getInstance(googleProject, masterInstanceZone, masterInstanceName) }
+      masterInstanceIp <- OptionT.fromOption { getInstanceIP(masterInstance) }
+    } yield masterInstanceIp
+
+    ipOpt.value
+  }
+
+  override def getClusterErrorDetails(operationName: String)(implicit executionContext: ExecutionContext): Future[Option[ClusterErrorDetails]] = {
+    val errorOpt = for {
+      operation <- OptionT.liftF[Future, Operation] { getOperation(operationName) }
+      done <- OptionT.pure { operation.getDone } if done == true
```

Comment:
only need `if done`

that said, done isn't used?

Reply:
Good call on `if done`.

I want to stop the flatMap if done == false so I'm filtering on it. Agreed this looks a little funky. Maybe could do instead:
```
_ <- OptionT.pure { operation.getDone }.filter(_ == true)
```